File: sop-sdk/sdk-python/response/BaseResponse.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
from UserDict import UserDict


class BaseResponse(UserDict):
    """返回类"""
    request_id = None
    code = None
    msg = None
    sub_code = None
    sub_msg = None

    def __init__(self, _dict=None, **kwargs):
        UserDict.__init__(self, _dict, **kwargs)

    def is_success(self):
        """是否成功

        :return: True,成功
        :rtype: bool
        """
        return self.sub_code is None

File: sop-sdk/sdk-go/response/BaseResponse.go
package response

import "encoding/json"

type IResponse interface {
	IsSuccess() bool
}

type BaseResponse struct {
	RequestId string `json:"request_id"`
	Code      string `json:"code"`
	Msg       string `json:"msg"`
	SubCode   string `json:"sub_code"`
	SubMsg    string `json:"sub_msg"`
}

func (resp BaseResponse) IsSuccess() bool {
	return len(resp.SubCode) == 0
}

func ConvertResponse(data []byte, ptr interface{})  {
	err := json.Unmarshal(data, ptr)
	if err != nil {
		panic(err)
	}
}
